Docs: add access and authorize log fields configuration settings (#906)
pomerium/documentation: content/docs/reference/authorize-log-fields.mdx

| `email` | The logged-in user's email address | Yes |
| `headers` | The complete set of HTTP request headers | No |
| \* `headers.{CustomHeaderName}` | An HTTP Request Header identified by the `HeaderName` | No |
| `host` | The HTTP request `:authority` or `Host` header value. Can be a domain name or IP address and may contain a port number (for example, `www.example.com`) | Yes |
| `id-token` | The user's ID token | No |
| `id-token-claims` | The user's ID token claims | No |
| `impersonate-email` | If [impersonating](/docs/capabilities/impersonation), the impersonated email | Yes |
| `impersonate-session-id` | If [impersonating](/docs/capabilities/impersonation), the impersonated session ID | Yes |
| `impersonate-user-id` | If [impersonating](/docs/capabilities/impersonation), the impersonated user ID | Yes |

> | `ip` | The user's IP address | Yes |

Reviewer: I think what we log for this value may vary depending on the [`xff_num_trusted_hops`](https://www.pomerium.com/docs/reference/the-number-of-trusted-hops) setting, so it might be good to link to that here.

### Find authorize logs

> To find authorize logs logged by the authorize service, filter your Pomerium logs for entries containing `"message": "authorize check"`:

Reviewer: ```suggestion
To find logs from the authorize service, filter your Pomerium logs for entries containing `"message": "authorize check"`:
```

### Display specific logs

If `authorize_log_fields` is not defined, all the available default fields will be logged.

> To display only a specific field, or multiple fields, add the desired field(s) to your configuration file.

Reviewer: ```suggestion
To log only a specific field, or multiple fields, add the desired field(s) to your configuration file.
```
